init/bash

```diff
@@ -1,7 +1,7 @@
 function show_msg {
   # only echo msg if EESSI_SILENT is unset
   msg=$1
-  if [[ ! -v EESSI_SILENT ]]; then
+  if [[ -z ${EESSI_SILENT+x} ]]; then
```

Comment:
I did a quick test, but it looks like this only works with single square brackets?

Reply:
I'm not seeing that:
```bash
alanc@~/EESSI/software-layer(allow_centos6_bash)$ source init/bash 
Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!
archdetect says x86_64/generic
Using x86_64/generic as software subdirectory.
Found Lmod configuration file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/.lmod/lmodrc.lua
Found Lmod SitePackage.lua file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/.lmod/SitePackage.lua
Using /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/modules/all as the directory to be added to MODULEPATH.
Using /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/generic/modules/all as the site extension directory to be added to MODULEPATH.
Initializing Lmod...
Prepending /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/modules/all to $MODULEPATH...
Prepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/generic/modules/all to $MODULEPATH...
Environment set up to use EESSI (2023.06), have fun!

{EESSI 2023.06} alanc@~/EESSI/software-layer(allow_centos6_bash)$ EESSI_SILENT=1 source init/bash 
Environment set up to use EESSI (2023.06), have fun!

{EESSI 2023.06} {EESSI 2023.06} alanc@~/EESSI/software-layer(allow_centos6_bash)$ echo $BASH_VERSION
5.2.21(1)-release

# BASH 3.2.57
{EESSI 2023.06} {EESSI 2023.06} alanc@~/EESSI/software-layer(allow_centos6_bash)$ ~/bash/bash-3.2.57/bash
bash: eval: line 24: syntax error near unexpected token `newline'
bash: eval: line 24: `Usage: lesspipe <file>'
bash: /usr/share/bash-completion/bash_completion: line 251: conditional binary operator expected
bash: /usr/share/bash-completion/bash_completion: line 251: syntax error near `exclude'
bash: /usr/share/bash-completion/bash_completion: line 251: `    if [[ -v exclude ]]; then'

alanc@~/EESSI/software-layer(allow_centos6_bash)$ EESSI_SILENT=1 source init/bash
Environment set up to use EESSI (2023.06), have fun!

{EESSI 2023.06} alanc@~/EESSI/software-layer(allow_centos6_bash)$ source init/bash
Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!
archdetect says x86_64/generic
Using x86_64/generic as software subdirectory.
Found Lmod configuration file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/.lmod/lmodrc.lua
Found Lmod SitePackage.lua file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/.lmod/SitePackage.lua
Using /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/modules/all as the directory to be added to MODULEPATH.
Using /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/generic/modules/all as the site extension directory to be added to MODULEPATH.
Initializing Lmod...
Prepending /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/generic/modules/all to $MODULEPATH...
Prepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/generic/modules/all to $MODULEPATH...
Environment set up to use EESSI (2023.06), have fun!

{EESSI 2023.06} {EESSI 2023.06} alanc@~/EESSI/software-layer(allow_centos6_bash)$ echo $BASH_VERSION
3.2.57(1)-release
```

Reply:
Oh, you're right, indeed. I think I may have tested it by executing a bash script instead of sourcing it. But I tried it again, and it does indeed work fine.